clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -556,8 +556,13 @@ class ScalarExprEmitter : public StmtVisitor<ScalarExprEmitter, mlir::Value> {
   mlir::Value VisitUnaryImag(const UnaryOperator *E) {
     llvm_unreachable("NYI");
   }
+
   mlir::Value VisitUnaryExtension(const UnaryOperator *E) {
-    llvm_unreachable("NYI");
+    TestAndClearIgnoreResultAssign();
```

Comment:
This diverges from upstream clang CodeGen. Actually I suspect it will crash in the following case:

```cpp
void test(int x) {
  return __extension__ x += 1;
}
```